linera-core/src/client/mod.rs

```diff
@@ -2460,6 +2371,7 @@ where
     ) -> Result<ClientOutcome<Option<ConfirmedBlockCertificate>>, ChainClientError> {
         let identity = self.identity().await?;
         let mut info = self.chain_info_with_manager_values().await?;
+        self.state().check_info_is_up_to_date(&info)?;
```

Comment:
I wonder if lines 2372-2382 can be rolled into a new function `update_chain_view(&self)`:
```rust
async fn update_chain_view(&self) -> Result<ChainInfoManager, ChainClientError> {
  let mut info = self.chain_info_with_manager_values().await?;
  self.state().check_info_is_up_to_date(&info)?;
  // If the current round has timed out, we request a timeout certificate and retry in
  // the next round.
  if let Some(round_timeout) = info.manager.round_timeout {
    if round_timeout <= self.storage_client().clock().current_time() {
      self.request_leader_timeout().await?;
        info = self.chain_info_with_manager_values().await?;
      }
   }
}
```

intuitively it seems like a correct thing to do, although not sure how reusable it would be. But it makes sense to me to group it like that. WDYT?

Reply:
It doesn't _download_ a chain info: It only loads the chain info from the local node, so that's quick.

The potentially slower part is requesting the timeout certificate and bringing the validators into the new round.